polylogue/cli/completion_engine.py

```
@@ -235,67 +531,99 @@ def _resolve_command_for_path(self, command: str) -> click.Command | None:
-                if branch_id and branch_id not in seen:
-                    entries.append(Completion(branch_id))
-                    seen.add(branch_id)
-        entries.sort(key=lambda c: c.value)
-        return entries
+        branch_ids = self._cached_state_list(
+            "branches",
+            lambda: sorted(
+                {
+                    branch_id
+                    for _, _, payload in self.env.conversations.iter_state()
+                    for branch_id in (payload.get("branches", {}) or {}).keys()
```

Ruff (0.15.9)

[warning] Use `key in dict` instead of `key in dict.keys()`

Remove `.keys()`

(SIM118)